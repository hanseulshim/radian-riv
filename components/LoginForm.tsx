const LoginForm: React.FC = () => {
  return <p>I am the login form</p>
}

export default LoginForm
