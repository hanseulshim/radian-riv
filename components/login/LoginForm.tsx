interface Props {
  showPasswordModal: () => void
}

const LoginForm: React.FC<Props> = ({ showPasswordModal }) => {
  return (
    <div className="login-form">
      <form>
        <div className="input-group">
          <input
            type="email"
            placeholder="Username..."
            name="Username"
            required
          />
          <label htmlFor="Username">Username</label>
        </div>
        <div className="input-group">
          <input
            type="password"
            placeholder="Password..."
            name="Password"
            required
          />
          <label htmlFor="Password">Password</label>
        </div>
        <button className="btn btn-primary login" type="submit">
          Login
        </button>
        <a href="#" onClick={() => showPasswordModal()}>
          Forgot Password
        </a>
        <span>or</span>
        <button className="btn btn-secondary btn-small register">
          Register for free
        </button>
      </form>
    </div>
  )
}

export default LoginForm
