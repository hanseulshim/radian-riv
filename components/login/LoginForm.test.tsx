import * as React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import LoginForm from './LoginForm'

describe('LoginForm', () => {
  let component
  beforeAll(() => {
    const props = {
      showPasswordModal: () => {},
      showRegisterModal: () => {}
    }
    component = shallow(<LoginForm {...props} />)
  })
  it('has login-form div', () => {
    expect(component.find('.login-form')).toHaveLength(1)
  })
  it('has a submit button', () => {
    expect(component.find('button').at(0).text()).toEqual('Login')
  })
  it('has a reset password', () => {
    expect(component.find('#reset-password-button').text()).toEqual(
      'Reset Your Password'
    )
  })
  it('has a register button', () => {
    expect(component.find('button').at(2).text()).toEqual('Register for free')
  })
  it('should match snapshot', () => {
    expect(toJson(component)).toMatchSnapshot()
  })
})
