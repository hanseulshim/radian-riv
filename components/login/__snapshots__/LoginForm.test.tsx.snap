// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm should match snapshot 1`] = `
<div
  className="login-form form"
>
  <Form
    alert={null}
    id="login"
    onSubmit={[Function]}
  >
    <Input
      error=""
      label="Username"
      onChange={[Function]}
      value=""
    />
    <Input
      error=""
      label="Password"
      onChange={[Function]}
      type="password"
      value=""
    />
    <button
      className="btn login-button"
      type="submit"
    >
      Login
    </button>
  </Form>
  <button
    className="btn btn-link"
    id="reset-password-button"
    onClick={[Function]}
  >
    Reset Your Password
  </button>
  <span>
    or
  </span>
  <button
    className="btn btn-secondary btn-small"
    onClick={[Function]}
  >
    Register for free
  </button>
</div>
`;
